src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +205,442 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-                frist column, result will include first column result, then
-                second and so on.
-        Returns:
-            pandas.DataFrame: a modified version of `same_smol` with a new
-                assigned data_col
+            to_scale: a data table where all records have been linked to EIA
+                plant-part list but they may be heterogeneous in its plant-part
+                granularities (i.e. some records could be of 'plant' plant-part
+                type while others are 'plant_gen' or 'plant_prime_mover').
+                All of the plant-part list columns need to be present in this
+                table.
+            cols_to_keep: columns from ``to_scale`` from the original dataset
+                that you want to show up in the output. These should not be
+                columns that show up in the ``ppl``.
+            ppl: the EIA plant-part list.
 
-        """
-        # we want to know the sum of the potential split_cols for each ferc1
-        # option
-        df_fgb = (
-            same_smol.loc[:, idx_cols_ferc1 + split_cols]
-            .groupby(by=idx_cols_ferc1, dropna=False)
-            .sum(min_count=1)
-            .reset_index()
-        )
-        df_w_tots = (
-            pd.merge(same_smol, df_fgb,
-                     on=idx_cols_ferc1,
-                     suffixes=("", "_fgb"))
-        )
-        # for each of the columns we want to split the frc data by
-        # generate the % of the total group, so we can split the data_col
-        new_data_col = self._get_clean_new_data_col(data_col)
-        df_w_tots[new_data_col] = pd.NA
-        for split_col in split_cols:
-            df_w_tots[f"{split_col}_pct"] = (
-                df_w_tots[split_col] / df_w_tots[f"{split_col}_fgb"])
-            # choose the first non-null option.
-            df_w_tots[new_data_col] = (
-                df_w_tots[new_data_col].fillna(
-                    df_w_tots[data_col] * df_w_tots[f"{split_col}_pct"]))
-
-        # merge in the newly generated split/assigned data column
-        df_final = pd.merge(
-            same_smol,
-            df_w_tots[idx_cols_ferc1 + ['record_id_eia_deprish',
-                                        new_data_col]].drop_duplicates(),
+        Returns:
+            A table which records coorespond to
+            :attr:``PlantPartScaler.plant_part`` (in this current
+            implementation: the records all coorespond to EIA generators!).
+            This is an intermediate table because the data columns from the
+            original dataset are duplicated.
+
+        """
+        # grab only the plant-part records that we are trying to scale to
+        ppl_part_df = ppl[ppl.plant_part == self.plant_part]
+        # convert the date to year start - this is necessary because the
+        # depreciation data is often reported as EOY and the ppl is always SOY
+        to_scale.loc[:, 'report_date'] = (
```

Comment:
This is mutating the input dataframe. Is that intentional / acceptable?

Reply:
this is mutating it. it is not ideal but feels necessary.